.github/workflows/verify-pr-commit.yml

```diff
@@ -653,4 +677,50 @@ jobs:
           WS_PROVIDER_URL="ws://127.0.0.1:9944" npm test
       - name: Stop Local Node
         if: always()
-        run: pkill ${{env.BIN_FILENAME}}
+        run: pkill ${{env.FREQUENCY_PROCESS_NAME}}
+
+  verify-genesis-state:
+    needs: build-binaries
+    name: Verify Genesis State
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set Env Vars
+        run: |
+          echo "BIN_DIR=target/$BUILD_PROFILE" >> $GITHUB_ENV
+          echo "EXPECTED_GENESIS_STATE_ROCOCO=0x000000000000000000000000000000000000000000000000000000000000000000e3495742b019f5ad49dff7de4040bc965b75eaf46769c24db1027d4ff86fc92703170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c11131400" >> $GITHUB_ENV
+          echo "EXPECTED_GENESIS_STATE_MAINNET=0x000000000000000000000000000000000000000000000000000000000000000000393a2a0f7778716d006206c5a4787cbf2ea3b26a67379b7a38ee54519d7fd4be03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c11131400" >> $GITHUB_ENV
+          echo "BIN_FILENAME_ROCOCO=frequency.rococo-pr" >> $GITHUB_ENV
+          echo "BIN_FILENAME_MAINNET=frequency.mainnet-pr" >> $GITHUB_ENV
+      - name: Check Out Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Download Binaries
+        uses: actions/download-artifact@v3
+        with:
+          name: artifacts-${{github.run_id}}
+          path: ${{env.BIN_DIR}}
+      - name: Set Binary Permissions
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          set -x
+          chmod 755 $BIN_FILENAME_ROCOCO
+          chmod 755 $BIN_FILENAME_MAINNET
+      - name: Test Rococo Genesis State
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          expected_genesis_state=${{env.EXPECTED_GENESIS_STATE_ROCOCO}}
+          echo "Expected genesis state: $expected_genesis_state"
+          actual_genesis_state=$(./${{env.BIN_FILENAME_ROCOCO}} export-genesis-state)
+          echo "Actual genesis state: $actual_genesis_state"
+          [[ $actual_genesis_state == $expected_genesis_state ]] || \
+            (echo "ERROR: The actual genesis state does not match the expected" && exit 10)
+      - name: Test Mainnet Genesis State
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          expected_genesis_state=${{env.EXPECTED_GENESIS_STATE_MAINNET}}
+          echo "Expected genesis state: $expected_genesis_state"
+          actual_genesis_state=$(./${{env.BIN_FILENAME_MAINNET}} export-genesis-state)
+          echo "Actual genesis state: $actual_genesis_state"
+          [[ $actual_genesis_state == $expected_genesis_state ]] || \
+            (echo "ERROR: The actual genesis state does not match the expected" && exit 10)
```

Comment:
Curious why `exit 10` instead of `exit 1`. GH Actions treats them the same afaik.